uxarray/grid/grid.py

```diff
@@ -2415,3 +2443,26 @@ def get_faces_at_constant_longitude(self, lon: float):
 
         faces = constant_lon_intersections_face_bounds(lon, self.face_bounds_lon.values)
         return faces
+
+    def get_faces_containing_point(self, point_xyz):
```

Comment:
We should change `point_xyz` to `point`, which can be either cartesian or spherical. Many times the user will want to query a spherical point, which we can convert to x, y, z.